source/applications/advanced/color_balance.py

```diff
@@ -0,0 +1,257 @@
+"""
+Balance color for RGB image.
+"""
+
+import datetime
+from pathlib import Path
+import numpy as np
+import matplotlib.pyplot as plt
+import zivid
+
+
+def _display_rgb(rgb, title):
+    """Display RGB image.
+
+    Args:
+        rgb: RGB image
+        title: Image title
+
+    Returns None
+
+    """
+    plt.figure()
+    plt.imshow(rgb)
+    plt.title(title)
+    plt.show(block=False)
+
+
+def _compute_mean_rgb(rgb, pixels):
+    """Compute mean RGB values.
+
+    Args:
+        rgb: RGB image
+        pixels: Number of central pixels for computation
+
+    Returns:
+        mean_red: Red channel mean value
+        mean_green: Green channel mean value
+        mean_blue: Blue channel mean value
+
+    """
+    height = np.shape(rgb)[0]
+    width = np.shape(rgb)[1]
+    red = rgb[
+        int((height - pixels) / 2) : int((height + pixels) / 2),
+        int((width - pixels) / 2) : int((width + pixels) / 2),
+        0,
+    ]
+    green = rgb[
+        int((height - pixels) / 2) : int((height + pixels) / 2),
+        int((width - pixels) / 2) : int((width + pixels) / 2),
+        1,
+    ]
+    blue = rgb[
+        int((height - pixels) / 2) : int((height + pixels) / 2),
+        int((width - pixels) / 2) : int((width + pixels) / 2),
+        2,
+    ]
+
+    mean_red = np.mean(np.reshape(red, -1, []))
+    mean_green = np.mean(np.reshape(green, -1, []))
+    mean_blue = np.mean(np.reshape(blue, -1, []))
+
+    return mean_red, mean_green, mean_blue
+
+
+def _rgb_from_3d_capture(camera):
```

Comment:
The input of this function should be a frame, not camera. It's confusing that you call this function and it will take the image for you.

Reply:
I will remove this function completely.